app/assets/javascripts/components/gtm-form-listener.js

```diff
@@ -0,0 +1,35 @@
+function GTMFormListener ($form, $dataLayer) {
+  this.$form = $form
+  this.dataLayer = $dataLayer || window.dataLayer
+}
+
+GTMFormListener.prototype.handleSubmit = function (event) {
+  var eventName = this.$form.dataset.gtm
+  var message = {}
+
+  var inputElements = this.$form.querySelectorAll('input:checked')
+  inputElements.forEach(function (element) {
+    if (element.value.length > 0) {
+      message = {}
+      message[eventName] = {}
+      message[eventName][element.name] = element.value
+      this.dataLayer.push(message)
+    }
+  }, this)
+}
+
+GTMFormListener.prototype.init = function () {
+  var $form = this.$form
+  if (!$form) {
+    return
+  }
+
+  $form.addEventListener('submit', function (e) {
+    this.handleSubmit(e)
+  }.bind(this), false)
+}
+
+var $gtmFormListener = document.querySelector('form[data-gtm]')
```

Comment:
should this be a querySelectorAll so that we can support multiple forms on a page?

Reply:
Fixed in bb39bebd9eb99562cc209abf2441f0513e3c18ea